app/lib/actions/admin-actions.ts

```diff
@@ -0,0 +1,113 @@
+'use server';
+import { revalidatePath } from 'next/cache';
+import { z } from 'zod';
+import prisma from '@/app/lib/prisma';
+
+const UserFormSchema = z.object({
+  name: z.string().trim().min(1, { message: 'Name is required' }),
+  email: z.string().email({ message: 'Invalid email address' }),
+  phone: z.string().trim().min(1, { message: 'Phone number is required' }),
+  role: z.enum(['RESIDENT', 'DOCTOR', 'ADMIN']),
```

Comment:
```suggestion
  role: z.nativeEnum(Role),
```

You can import the `Role` type from the Prisma client I think. The reason we want to do this is because if we ever change the Role enum in our Prisma schema, we won't have to manually update `['RESIDENT', 'DOCTOR', 'ADMIN']` here.